packit_service/worker/helpers/testing_farm.py

```diff
@@ -60,40 +61,72 @@ class CommentArguments:
         logger.debug(f"Parsing comment -> {comment}")
         logger.debug(f"Used command prefix -> {command_prefix}")
 
-        match = re.search(
-            r"^" + re.escape(command_prefix) + r"\s(?P<packit_command>\S+)", comment
-        )
+        # Match the command prefix and extract the rest of the comment
+        match = re.match(r"^" + re.escape(command_prefix) + r"\s+(.*)", comment)
         if match:
-            self.packit_command = match.group("packit_command")
-            logger.debug(f"Parsed packit_command: {self.packit_command}")
+            arguments_str = match.group(1)
+        else:
+            # If the command prefix is not found, nothing to parse
+            logger.debug("Command prefix not found in the comment.")
+            return
 
-        match = re.search(r"(--identifier|--id|-i)[\s=](?P<identifier>\S+)", comment)
-        if match:
-            self.identifier = match.group("identifier")
-            logger.debug(f"Parsed test argument -> identifier: {self.identifier}")
+        # Use shlex to split the arguments string into a list
+        args_list = shlex.split(arguments_str)
+        logger.debug(f"Arguments list after shlex splitting: {args_list}")
 
-        match = re.search(r"--labels[\s=](?P<labels>\S+)", comment)
-        if match:
-            self.labels = match.group("labels").split(",")
-            logger.debug(f"Parsed test argument -> labels: {self.labels}")
+        # Set up argparse
+        parser = argparse.ArgumentParser()
+        parser.add_argument("packit_command")
+        parser.add_argument("--identifier", "--id", "-i")
+        parser.add_argument("--labels")
+        parser.add_argument("--env", action="append")  # Allows multiple --env arguments
 
-        match = re.search(r"(?P<pr_arg>[^/\s]+/[^#]+#\d+)", comment)
-        if match:
-            self.pr_argument = match.group("pr_arg")
-            logger.debug(f"Parsed test argument -> pr_argument: {self.pr_argument}")
+        # Parse known arguments
+        try:
+            args, unknown_args = parser.parse_known_args(args_list)
+            logger.debug(f"Parsed known args: {args}")
+            logger.debug(f"Unknown args: {unknown_args}")
+        except argparse.ArgumentError as e:
+            logger.error(f"Argument parsing error: {e}")
+            return
+
+        # Assign the parsed arguments to the class attributes
+        self.packit_command = args.packit_command
+        logger.debug(f"Parsed packit_command: {self.packit_command}")
+
+        self.identifier = args.identifier
+        logger.debug(f"Parsed identifier: {self.identifier}")
+
+        if args.labels:
+            self.labels = args.labels.split(",")
+            logger.debug(f"Parsed labels: {self.labels}")
+
+        if args.env:
+            # Parse envs into dictionary
+            self.envs = dict(env.split("=") for env in args.env)
+            logger.debug(f"Parsed envs: {self.envs}")
```

Comment:
I think here we should add some error handling for bad input (not `key=value` format)

Reply:
I did some changes, please take a look :)